net-irc/znc/znc-9999.ebuild

```diff
@@ -2,163 +2,170 @@
+PLOCALES="ru"
+CMAKE_MAKEFILE_GENERATOR=ninja
 
-MY_PV=${PV/_/-}
+inherit cmake-utils l10n python-single-r1 readme.gentoo-r1 systemd user
+
+GTEST_VER="1.8.0"
+GTEST_URL="https://github.com/google/googletest/archive/release-${GTEST_VER}.tar.gz -> googletest-release-${GTEST_VER}.tar.gz"
 DESCRIPTION="An advanced IRC Bouncer"
 
-inherit git-r3
-EGIT_REPO_URI=${EGIT_REPO_URI:-"git://github.com/znc/znc.git"}
-SRC_URI=""
-KEYWORDS=""
+if [[ ${PV} == *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI=${EGIT_REPO_URI:-"git://github.com/znc/znc.git"}
+	SRC_URI=""
+	KEYWORDS=""
+	SWIG_DEPEND="
+		perl? ( >=dev-lang/swig-3.0.0 )
+		python? ( >=dev-lang/swig-3.0.0 )
+	"
+else
+	SRC_URI="
+		http://znc.in/releases/archive/${P}.tar.gz
+		test? ( ${GTEST_URL} )
+	"
+	KEYWORDS="~amd64 ~arm ~x86"
+	SWIG_DEPEND=""
+fi
 
 HOMEPAGE="http://znc.in"
 LICENSE="Apache-2.0"
 SLOT="0"
-IUSE="daemon debug ipv6 libressl perl python ssl sasl tcl"
+IUSE="daemon +ipv6 +icu libressl nls perl python +ssl sasl tcl test +zlib"
 
-REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} icu )"
 
 RDEPEND="
-	dev-libs/icu:=
-	sys-libs/zlib
-	perl? ( >=dev-lang/perl-5.10 )
+	icu? ( dev-libs/icu:= )
+	nls? ( dev-libs/boost[nls] )
+	perl? ( >=dev-lang/perl-5.10:= )
 	python? ( ${PYTHON_DEPS} )
 	sasl? ( >=dev-libs/cyrus-sasl-2 )
 	ssl? (
-		!libressl? ( dev-libs/openssl:0 )
-		libressl? ( dev-libs/libressl )
+		!libressl? ( dev-libs/openssl:0= )
+		libressl? ( dev-libs/libressl:0= )
 	)
 	tcl? ( dev-lang/tcl:0= )
+	zlib? ( sys-libs/zlib )
 "
-DEPEND="${RDEPEND}
+DEPEND="
+	${RDEPEND}
+	${SWIG_DEPEND}
+	nls? ( sys-devel/gettext )
 	virtual/pkgconfig
-	perl? (
-		>=dev-lang/swig-2.0.12
-	)
-	python? (
-		>=dev-lang/swig-2.0.12
-	)
 "
 
-S=${WORKDIR}/${PN}-${MY_PV}
-
-CONFDIR="/var/lib/znc"
-
 pkg_setup() {
 	if use python; then
 		python-single-r1_pkg_setup
 	fi
 	if use daemon; then
 		enewgroup ${PN}
-		enewuser ${PN} -1 -1 /dev/null ${PN}
+		enewuser ${PN} -1 -1 /var/lib/${PN} ${PN}
+		# The home directory was previously set to /dev/null
+		# This caused a bug with the systemd unit
+		# https://bugs.gentoo.org/521916
+		esethome ${PN} /var/lib/${PN}
 	fi
 }
 
 src_prepare() {
-	AT_M4DIR="${S}/m4" \
-		eautoreconf
-
-	# build system quirk, it does not define AM_INIT_AUTOMAKE, nor
-	# does it have Makefile.am in the root dir, but we need '--add-missing'
-	automake --add-missing
+	l10n_find_plocales_changes "${S}/src/po" "${PN}." '.po'
+	remove_locale() {
+		# Some languages can miss some modules
+		rm src/po/${PN}.${1}.po modules/po/*.${1}.po || true
+	}
+	l10n_for_each_disabled_locale_do remove_locale
+	cmake-utils_src_prepare
 }
 
 src_configure() {
-	econf \
-		$(use_enable debug) \
-		$(use_enable ipv6) \
-		$(use_enable perl) \
-		$(use python && echo "--enable-python=python3") \
-		$(use_enable sasl cyrus) \
-		$(use_enable ssl openssl) \
-		$(use_enable tcl tcl)
+	local want_swig
+	if [[ ${PV} == *9999* ]]; then
+		want_swig=yes
```

Comment:
Tarballs include files generated by SWIG, so they don't need it.
However, thinking about it more, to allow more flexibility with `/etc/portage/patches/net-irc/znc`, probably SWIG should be used for releases too, not only for live.